src/Unit.hs

```diff
@@ -28,7 +26,7 @@ import Numeric
 
 -- * Units
 
-class Applicative u => Unit u where
+class Applicative u => Unit (dim :: * -> *) u | u -> dim where
```

Comment:
Adding a dimension to `Unit` means that we can now ensure that conversions are only made between units of the same dimension. We don’t risk converting `Metres` to `Seconds` in some sort of space-time _eversion_ or something.